src/controllers/DialogController.ts

```diff
@@ -0,0 +1,23 @@
+import { Request as ExpressRequest, Response } from "express";
+import { Schema } from "mongoose";
+
+import CreateDialogService from "../services/CreateDialogService";
+import ConsultDialogService from "../services/ConsultDialogService";
+
+interface Request extends ExpressRequest {
```

Comment:
Could you explain to me why this is here?

Reply:
We're accessing the ```userId``` property that was created by the ```AuthMiddleware```. Since it's a customization of the ```Request``` interface the typescript intellisense cannot access this property. So, we need to create an interface that extends the ```Request``` from Express

Reply:
Need to use this in the controller and in the services as well?

Reply:
Yes because we are passing the ```Response``` object from ```express.Router``` to the service inside each controller method. Types must be the same.